reports/reports.service.js

```diff
@@ -0,0 +1,46 @@
+const Report = require('./reports.model');
+const schema = require('./report.schema');
+const AppError = require('../util/appError');
+const catchAsync = require('../util/catchAsync');
+let Ajv = require('ajv');
+let ajv = new Ajv({ allErrors: true });
+
+exports.reportService = {
+  create() {
+    return catchAsync(async (req, res, next) => {
+      const data = {
+        resourceId: req.body.resourceId,
+        reporterId: req.body.reporterId,
+        type: req.body.type
+      };
+
+      let validate = ajv.compile(schema);
+      let valid = validate(data);
+      if (!valid) {
+        let errorsMesssage = '';
+        const errors = validate.errors;
+        console.log(errors);
+        for (let error of errors) {
+          errorsMesssage += error.dataPath + ': ' + error.message + ', ';
+        }
+        return next(new AppError(errorsMesssage, 400, errors));
+      }
+
+      if (req.body.reporterId.toString() === req.user.mongouser._id.toString())
```

Comment:
This isn't checking if the post being reported is owned by the reporter id, 
actually the reporter id must equal the id of the currently logged in user.
so this check must be done, but to allow the user to report not to stop him from reporting